src/test/sql/sql_translator_test.cpp

```diff
@@ -1865,16 +1868,44 @@ TEST_F(SQLTranslatorTest, ShowColumns) {
   const auto actual_lqp = compile_query("SHOW COLUMNS int_float");
 
   // clang-format off
-  const auto stored_table_node = StoredTableNode::make(MetaTableManager::META_PREFIX + "columns");
-  const auto table_name_column = stored_table_node->get_column("table_name");
+  const auto meta_table = Hyrise::get().meta_table_manager.generate_table("columns");
+  const auto static_table_node = StaticTableNode::make(meta_table);
+  const LQPColumnReference table_name_column{static_table_node, meta_table->column_id_by_name("table_name")};
+
   const auto expected_lqp =
       PredicateNode::make(equals_(table_name_column, "int_float"),
-        stored_table_node);
+        static_table_node);
   // clang-format on
 
   EXPECT_LQP_EQ(actual_lqp, expected_lqp);
 }
 
+TEST_F(SQLTranslatorTest, SelectMetaTable) {
+  const auto actual_lqp = compile_query("SELECT * FROM " + MetaTableManager::META_PREFIX + "tables");
+
+  const auto meta_table = Hyrise::get().meta_table_manager.generate_table("tables");
+  const auto expected_lqp = StaticTableNode::make(meta_table);
+
+  EXPECT_LQP_EQ(actual_lqp, expected_lqp);
+}
+
+TEST_F(SQLTranslatorTest, SelectMetaTableSubquery) {
+  const auto actual_lqp = compile_query("SELECT table_name FROM (SELECT table_name, column_count FROM " +
+                                        MetaTableManager::META_PREFIX + "tables) as subquery");
+
+  const auto meta_table = Hyrise::get().meta_table_manager.generate_table("tables");
+  const auto static_table_node = StaticTableNode::make(meta_table);
+
+  const LQPColumnReference table_name_column{static_table_node, meta_table->column_id_by_name("table_name")};
+  const LQPColumnReference column_count_column{static_table_node, meta_table->column_id_by_name("column_count")};
+
+  const auto expected_subquery_lqp =
+      ProjectionNode::make(expression_vector(table_name_column, column_count_column), static_table_node);
+  const auto expected_lqp = ProjectionNode::make(expression_vector(table_name_column), expected_subquery_lqp);
+
+  EXPECT_LQP_EQ(actual_lqp, expected_lqp);
+}
+
```

Comment:
Do we have a test where the same meta table is used twice and where we test that both StaticTableNodes point to the same table?

Reply:
we have a little conflict here with the copyable flag of storedTableNodes. Right now, the only situation, where this could happen is in a where clause but since where clauses copy nodes, they are currently not allowed for meta tables. other use cases such as unions are not yet implemented. 
imo, we should drop the copyable flag, since copying a meta table within one query isn't a problem anyway

Reply:
What do you mean by where clauses are not allowed for meta tables? `SELECT * FORM meta_segments WHERE table_name = 'lineitem'` does work.

Reply:
yes. but if the where clause contains a subquery, it's not allowed anymore. E.g.
`SELECT * FROM meta_tables WHERE table_name != (SELECT table_name from meta_tables LIMIT 1)`

Reply:
anyway, there is now a test that covers the case where a meta table is used twice but generated only once. somehow, i thought we need a where clause for this to happen but a self join causes this as well

Reply:
Great.